src/Renci.SshNet/Session.cs

```diff
@@ -1103,10 +1113,105 @@ internal void SendMessage(Message message)
                 //
                 // the server will use it to verify the data integrity, and as such the order in
                 // which messages are sent must follow the outbound packet sequence number
+#if NET6_0_OR_GREATER
+                _ = Interlocked.Increment(ref _outboundPacketSequence);
+#else
                 _outboundPacketSequence++;
+#endif
```

Comment:
Interlocked is not needed here

```suggestion
                _outboundPacketSequence++;
```